### PR TITLE
Updating development setup steps

### PR DESCRIPTION
**Pull request by apoorvajagtap**
Enhancing the setup steps in `CONTRIBUTING.md` & some configurations in `Dockerfile-dev` (could be useful for community contributors).

### CODE REVIEW

**Review by coreydaley — COMMENTED** [review]
Looks good, just a couple of comments

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -47,7 +54,7 @@ $ make build-devel-image IMAGE=<MYREPO>/<MYIMAGE> TAG=<MYTAG>
 or if you are using `buildah`:
 
 ```
-$ buildah bud -t <MYREPO>/<MYIMAGE>:<MYTAG> -f Dockerfile.dev .
+$ buildah bud -t <MYREPO>/<MYIMAGE>:<MYTAG> -f Dockerfile-dev .
```
[the remark this answers is not in the document]
Comment (apoorvajagtap) [reply CONTRIBUTING.md:57]: I think it'll make more sense to rename the filename with a dot. Would need to just modify the Makefile (in addition)

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -24,7 +24,7 @@ There are three types of channels you can obtain the installer from:
 2. `4.x.0-nightly` - nightly development releases, with payloads published to quay.io.
 3. `4.x.0-ci` - bleeding-edge releases published to the OpenShift CI imagestreams.
 
-**Note**: Installs from the `4.x.0-ci` channel require a pull secret to `registry.svc.ci.openshift.org`, which is only available to Red Hat OpenShift developers.
+**Note**: Installs from the `4.x.0-ci` channel require a pull secret to `registry.ci.openshift.org`, which is only available to Red Hat OpenShift developers.
```
[the remark this answers is not in the document]
Comment (apoorvajagtap) [reply CONTRIBUTING.md:27]: Definitely! on it.

**Comment by coreydaley** [comment]
/assign @bparees 
Can you please set the `bugzilla/valid-bug` label? Seems a bit overkill to have a bug for this update.

**Comment by bparees** [comment]
> Can you please set the bugzilla/valid-bug label? Seems a bit overkill to have a bug for this update.

This is a PR against master, so it doesn't need that...(didn't need the other backport-related labels either)

**Comment by coreydaley** [comment]
@bparees It was not merging because it prow said it needed it. If it does it again I will capture a screen shot.

**Comment by apoorvajagtap** [comment]
@coreydaley observed some typos and corrected those in the recent commit. Could you please give another lgtm?

**Comment by bparees** [comment]
right now it says

```
 Not mergeable. Needs bugzilla/valid-bug, lgtm labels OR Needs lgtm label. 
 ```
 
 not sure if that's what you saw  :shrug:

**Pull request merged**